### PR TITLE
Allow Specifying a different version for base and maps on android

### PR DESCRIPTION
**Pull request by Titozzz**
### Does any other open PR do the same thing?

No

### What issue is this PR fixing?

#2584 

### How did you test this PR?

This allows android to specify a custom version for 
`com.google.android.gms : play-services-maps`
and
`com.google.android.gms : play-services-base
`
by specifying 

```
	googlePlayServicesVersion = "16.0.1"
    googlePlayServicesAuthVersion = "16.0.1"// this is how react-native-google-signin does 
    googlePlayServicesMapsVersion = "16.0.0"
```

for example I had to use 16.0.1 for base but maps does not exist for that version
https://mvnrepository.com/artifact/com.google.android.gms/play-services-maps

It should automatically fallback to googlePlayServicesVersion value if googlePlayServicesMapsVersion is not specified

### CODE REVIEW

File: lib/android/build.gradle [hunk 1]
```diff
@@ -40,11 +41,13 @@ android {
 }
 
 dependencies {
-  def googlePlayServicesVersion = rootProject.hasProperty('googlePlayServicesVersion')  ? rootProject.googlePlayServicesVersion : DEFAULT_GOOGLE_PLAY_SERVICES_VERSION
-  def androidMapsUtilsVersion   = rootProject.hasProperty('androidMapsUtilsVersion')    ? rootProject.androidMapsUtilsVersion   : DEFAULT_ANDROID_MAPS_UTILS_VERSION
+  def googlePlayServicesVersion = rootProject.hasProperty('googlePlayServicesVersion') ? rootProject.googlePlayServicesVersion : DEFAULT_GOOGLE_PLAY_SERVICES_VERSION
+  // googlePlayServicesMapsVersion > googlePlayServicesVersion > DEFAULT_GOOGLE_PLAY_SERVICES_MAPS_VERSION (16.0.0)
```
Comment (rborn) [inline lib/android/build.gradle]: do we need this commented code?

**Comment by rborn** [comment]
@Titozzz could you check with @yeomann maybe use `getSafeExt` like in https://github.com/react-community/react-native-maps/pull/2398/files ?

Thnx

File: lib/android/build.gradle (hunk 1 above)
Comment (Titozzz) [reply lib/android/build.gradle]: It's not code, it was a comment so that people understand the line below. It can remove it completely

**Comment by Titozzz** [comment]
Well I could import his or inside of monebif you want to, or wait for his to be merged

File: lib/android/build.gradle (hunk 1 above)
Comment (rborn) [reply lib/android/build.gradle]: Maybe you could make then more clear (more descriptive) or delete it, your call 😄

**Comment by rborn** [comment]
@Titozzz let's wait for @yeomann to reply, and if not maybe you can import the PR in yours and push again. 🤗

**Comment by Titozzz** [comment]
Sure thing 😄

**Comment by yeomann** [comment]
hey guys, sorry for delay. sure thing mate @Titozzz go ahead 😃

**Comment by Titozzz** [comment]
Will do :)

**Comment by Titozzz** [comment]
@rborn Done ✅

**Comment by rborn** [comment]
@Titozzz sorry man, vacations + very busy, done now ❤️

**Comment by Titozzz** [comment]
Don't worry ❤️ This was part of emptying my github gigantic pile of notifications after vacations :D !

Thanks ! you can close #2584

**Pull request merged**